contracts/MyStrategy.sol

```diff
@@ -453,18 +497,39 @@ contract MyStrategy is BaseStrategy, ReentrancyGuardUpgradeable {
         emit RewardsCollected(token, amount);
     }
 
-    /// @dev Send the BADGER token to the badgerTree
-    function _sendBadgerToTree(uint256 amount) internal {
-        // Transfer to tree without taking any fee
-        IERC20Upgradeable(BADGER).safeTransfer(IVault(vault).badgerTree(), amount);
-        emit TreeDistribution(BADGER, amount, block.number, block.timestamp);
+    /// @dev Takes a fee on the token and sends remaining to the given briber recepient
+    function _sendTokenToBriber(address token, address recepient, uint256 amount) internal {
+        // Process redirection fee
+        uint256 redirectionFee = amount.mul(redirectionFees[recepient]).div(MAX_BPS);
+        if (redirectionFee > 0) {
+            IERC20Upgradeable(token).safeTransfer(IVault(vault).treasury(), redirectionFee);
+            emit RedirectionFee(
+                IVault(vault).treasury(),
+                token,
+                redirectionFee,
+                block.number,
+                block.timestamp
+            );
+        }
+
+        // Send remaining to bribe recepient
+        // NOTE: Calculating instead of checking balance since there could have been an 
+        // existing balance on the contract beforehand
+        IERC20Upgradeable(token).safeTransfer(recepient, amount.sub(redirectionFee));
```

Comment:
do we need a non-zero check before `safeTransfer`?

Reply:
This function will only be called if there was an amount claimed so `amount` will never be 0. The only reason the subtraction would lead to 0 is if the fee == MAX_BPS (100%). This shouldn't be the case but, since the fee can be set to this amount, it will indeed be good to protect against it - will fix.